frontend/src/Web3Provider.js

```diff
@@ -8,33 +8,41 @@ class Web3Provider extends Component {
     this.state = {
       web3: null,
       accounts: [],
-      networkId: null
+      networkId: null,
+      allFetched: null
```

Comment:
I added this mainly for the tests, but I think it may be useful for the parent component too (for example, to avoid showing the `No unlocked accounts` for some milliseconds).